Remove 'set_pipeline' experimental warnings

analytically opened the pull request:
The set_pipeline feature is now considered stable and widely used, so remove the warning messages that marked it as experimental. This affects both the self-referential pipeline syntax and team specification warnings.

# Release Note

* Removed "experimental" warnings from the `set_pipeline` step

taylorsilva reviewed (approved):
Makes sense, it's not going anywhere or changing behaviour. It's pretty settled.

taylorsilva reviewed (changes requested):
Forgot to remove some tests:
```
Summarizing 2 Failures:

  [FAIL] SetPipelineStep when file is configured when pipeline file is good when team is configured when team exists when the target team is the current team [It] should print an experimental message

  github.com/concourse/concourse/atc/exec/set_pipeline_step_test.go:519

  [FAIL] SetPipelineStep when file is configured when pipeline file is good when set-pipeline self [It] should print an experimental message

  github.com/concourse/concourse/atc/exec/set_pipeline_step_test.go:443
```

analytically committed:
Remove 'set_pipeline' experimental warnings

The set_pipeline feature is now considered stable and widely used, so remove the warning messages that marked it as experimental. This affects both the self-referential pipeline syntax and team specification warnings.

Signed-off-by: Mathias Bogaert <mathias.bogaert@gmail.com>

analytically commented:
Removed the test checks.
